packages/core/contracts/optimistic-assertor/implementation/examples/DataAsserter.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../../interfaces/OptimisticAssertorInterface.sol";
+import "../../../common/implementation/AncillaryData.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+// This contract allows assertions on any form of data to be made using the UMA Optimistic Assertor and stores the
+// proposed value so that it may be retrieved on chain. It only enables one assertion per pair of dataId and asserter
+// address. The dataId is intended to be an arbitrary value that uniquely identifies a specific piece of information in
+// the consuming contract and is replaceable. Similarly, any data structure can be used to replace the asserted data.
+contract DataAsserter {
+    using SafeERC20 for IERC20;
+    IERC20 public immutable defaultCurrency;
+    OptimisticAssertorInterface public immutable oa;
+    uint256 public constant assertionLiveness = 7200;
+    bytes32 public immutable defaultIdentifier;
+
+    struct DataAssertion {
+        bytes32 dataId; // This could be an arbitrary data type.
```

Comment:
Does the comment mean the data type of this property would be changed?

Reply:
Yes, but I think we could remove it as this could represent the id of any kind of data already